app/src/test/java/waiter/MessengerTest.java

```diff
@@ -21,23 +27,25 @@ void setUp() {
         clientHasDisconnected = null;
     }
 
-    @Test
-    void echosSingleMessage() throws IOException {
-        String[] clientInputs = {"foo", clientHasDisconnected};
+    @Property(tries = 5)
+    void echosSingleMessage(@ForAll @AlphaChars @NotBlank String userInput) throws IOException {
+        String[] clientInputs = {userInput, clientHasDisconnected};
         ClientConnectionMock clientConnectionMock = new ClientConnectionMock(clientInputs);
 
         new Messenger(new EchoProtocolMock()).transport(clientConnectionMock);
 
-        assertArrayEquals(new String[]{"foo"}, clientConnectionMock.echoedInputs);
+        assertArrayEquals(new String[]{userInput}, clientConnectionMock.echoedInputs);
     }
 
-    @Test
-    void echosManyMessages() throws IOException {
-        String[] clientInputs = {"foo", "bar", "baz", clientHasDisconnected};
+    @Property(tries = 5)
+    void echosManyMessages(@ForAll @Size(3) List<@AlphaChars @StringLength(min = 1, max = 10) String> userInputs) throws IOException {
```

Comment:
Is the length of the String relevant here? Do we see issues if we allow this length to be unbounded?